csrc/fusion.cpp

```diff
@@ -797,28 +797,37 @@ void Fusion::aliasOutputToInput(
     }
   }
 
-  NVF_ERROR(
-      isAliasCompatible(input, output),
-      "The input and output values are not alias-compatible.");
-  // Let integration hide any output that wasn't a fusion output when
-  // `aliasOutputToInput` was called. For example, running mean and var for
-  // batch norm.
-  io_alias_[output] = AliasInfo{
-      .type = type,
-      .aliased_io = input,
-      .hide_output = !output->isFusionOutput()};
+  // AllocationType::Evaluate
+  if (!input) { // Non-aliased outputs marked for expression evaluator.
+    NVF_CHECK(
+        output->isFusionOutput(),
+        "Only fusion outputs can be expression evaluated.");
+    io_alias_[output] =
+        AliasInfo{.type = type, .aliased_io = nullptr, .hide_output = false};
+  }
 
-  // TODO: output should be marked at the end of fusion definition #1488
-  if (!output->isFusionOutput()) {
-    addOutput(output);
+  else { // Pointer Arithmetic Outputs
+    NVF_ERROR(
+        isAliasCompatible(input, output),
```

Comment:
Nit: if it helps to simplify, feel free to call this only for `ReuseBuffer`, which is what it was written for anyway. I tried to remove some checks there to accommodate pointer-arithmetic aliases, but these checks are minimal and can break in the future, e.g., we could bitcast a tensor to a different data type.